core/src/toga/window.py

```diff
@@ -465,21 +471,33 @@ def visible(self, visible: bool) -> None:
     ######################################################################
 
     @property
-    def full_screen(self) -> bool:
-        """Is the window in full screen mode?
+    def state(self) -> WindowState:
+        """The current state of the window.
 
-        Full screen mode is *not* the same as "maximized". A full screen window
-        has no title bar, toolbar or window controls; some or all of these
-        items may be visible on a maximized window. A good example of "full screen"
-        mode is a slideshow app in presentation mode - the only visible content is
-        the slide.
+        When the window is in transition, then this will return the state it
+        is transitioning towards, instead of the actual instantaneous state.
         """
-        return self._is_full_screen
-
-    @full_screen.setter
-    def full_screen(self, is_full_screen: bool) -> None:
-        self._is_full_screen = is_full_screen
-        self._impl.set_full_screen(is_full_screen)
+        # There are 2 types of window states that we can get from the backend:
+        # * The instantaneous state -- Used internally on implementation side
+        # * The in-progress state -- Used for same state checking on the core
+        #                            and for the public API.
+        return self._impl.get_window_state(in_progress_state=True)
+
+    @state.setter
+    def state(self, state: WindowState) -> None:
+        if not self.visible:
+            raise ValueError("Window state of a hidden window cannot be changed.")
```

Comment:
As above - this should be a documented RuntimeError.

Reply:
I have added docstrings to `state`.